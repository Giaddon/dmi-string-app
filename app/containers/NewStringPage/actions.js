import {
  CHANGE_STRING,
  ADD_STRING,
  ADD_STRING_SUCCESS,
  ADD_STRING_FAILURE,
} from './constants';

/**
 * Dispatched when the user makes a change to the new string input.
 *
 * @param {string} newString The submitted string
 *
 * @return {object} An action object with a type of CHANGE_STRING and the new string.
 */

export function changeString(newString) {
  return {
    type: CHANGE_STRING,
    newString,
  };
}

/**
 * Dispatched when the user submits a new string for the server.
 *
 * @return {object} An action object with a type of ADD_STRING.
 */

export function addString() {
  return { type: ADD_STRING };
}
/**
 * Dispatched when the new string is added to ther server successfully.
 *
 * @param {string} newString The newly-added string sent from the server
 *
 * @return {object} An action object with a type of ADD_STRINGS_SUCCESS and the server response.
 */
export function addStringSuccess(newString) {
  return { type: ADD_STRING_SUCCESS, newString };
}
/**
 * Dispatched when there's an error adding a new string to the server.
 *
 * @param {object} error The error
 *
 * @return {object} An action object with a type of ADD_STRING_FAILURE and the error.
 */
export function addStringFailure(error) {
  return { type: ADD_STRING_FAILURE, error };
}
