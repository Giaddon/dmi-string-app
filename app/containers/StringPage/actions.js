import {
  GET_STRINGS,
  GET_STRINGS_SUCCESS,
  GET_STRINGS_FAILURE,
} from './constants';

/**
 * Requests string data from the server.
 *
 * @return {object} An action object with a type of GET_STRINGS
 */
export function getStrings() {
  return { type: GET_STRINGS };
}

/**
 * Dispatched when string data recieved from the server.
 *
 * @param  {array} strings An array of strings recieved from the server.
 *
 * @return {object} An action object with a type of GET_STRINGS_SUCCESS and the string array
 */
export function getStringsSuccess(strings) {
  return { type: GET_STRINGS_SUCCESS, strings };
}

/**
 * Dispatched when there's an error retrieving string data from the server.
 *
 * @param {object} error The error
 *
 * @return {object} An action object with a type of GET_STRINGS_FAILURE and the error.
 */
export function getStringsFailure(error) {
  return { type: GET_STRINGS_FAILURE, error };
}
