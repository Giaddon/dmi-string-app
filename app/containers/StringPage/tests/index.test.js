/**
 * Test the StringPage.
 */

import React from 'react';
import { render } from 'react-testing-library';
import { IntlProvider } from 'react-intl';
import { Provider } from 'react-redux';
import { browserHistory } from 'react-router-dom';
import { StringPage, mapDispatchToProps } from '../index';
import { getStrings } from '../actions';
import configureStore from '../../../configureStore';

describe('<StringPage />', () => {
  let store;

  beforeAll(() => {
    store = configureStore({}, browserHistory);
  });

  it('should render and match the snapshot', () => {
    const {
      container: { firstChild },
    } = render(
      <Provider store={store}>
        <IntlProvider locale="en">
          <StringPage
            loading
            onPageLoad={() => {}}
            alert={{ display: false, message: '' }}
          />
        </IntlProvider>
      </Provider>,
    );
    expect(firstChild).toMatchSnapshot();
  });

  it('should fetch the strings on mount', () => {
    const loadSpy = jest.fn();
    render(
      <Provider store={store}>
        <IntlProvider locale="en">
          <StringPage
            onPageLoad={loadSpy}
            loading
            alert={{ display: false, message: '' }}
          />
        </IntlProvider>
      </Provider>,
    );
    expect(loadSpy).toHaveBeenCalled();
  });

  describe('mapDispatchToProps', () => {
    describe('onPageLoad', () => {
      it('should be injected', () => {
        const dispatch = jest.fn();
        const result = mapDispatchToProps(dispatch);
        expect(result.onPageLoad).toBeDefined();
      });

      it('should dispatch getStrings when called', () => {
        const dispatch = jest.fn();
        const result = mapDispatchToProps(dispatch);
        result.onPageLoad();
        expect(dispatch).toHaveBeenCalledWith(getStrings());
      });
    });
  });
});
