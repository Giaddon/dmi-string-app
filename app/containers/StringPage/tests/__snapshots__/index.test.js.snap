// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<StringPage /> should render and match the snapshot 1`] = `
<div>
  <div
    class="LoadingIndicator__LoadingDiv-sc-8mhje4-0 ilyFRF"
  >
    <h2>
      Loading...
    </h2>
  </div>
</div>
`;
